flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -2213,12 +2211,24 @@ public final JobClient executeAsync() throws Exception {
      */
     @PublicEvolving
     public JobClient executeAsync(String jobName) throws Exception {
-        Preconditions.checkNotNull(jobName, "Streaming Job name should not be null.");
         final StreamGraph streamGraph = getStreamGraph();
-        streamGraph.setJobName(jobName);
+        setJobNameForJobGraph(streamGraph, jobName);
```

Comment:
IMO, The logic is simple and clear, and there is no need to introduce a new function.

Reply:
fixed